src/main/java/uk/gov/hmcts/reform/sscs/bulkscancore/handlers/CcdCallbackHandler.java

```diff
@@ -211,15 +215,22 @@ && isNotBlank(callback.getCaseDetails().getCaseData().getAppeal().getBenefitType
             }
 
             String processingVenue = sscsDataHelper.findProcessingVenue(appeal.getAppellant(), appeal.getBenefitType());
+
             if (isNotBlank(processingVenue)) {
                 callback.getCaseDetails().getCaseData().setProcessingVenue(processingVenue);
                 if (caseAccessManagementFeature) {
                     CourtVenue courtVenue = refDataService.getVenueRefData(processingVenue);
-
                     if (courtVenue != null) {
-                        callback.getCaseDetails().getCaseData().setCaseManagementLocation(CaseManagementLocation.builder()
-                            .baseLocation(courtVenue.getEpimsId())
-                            .region(courtVenue.getRegionId()).build());
+                        String rpcEpimsId = rpcVenueService.retrieveRpcEpimsIdForAppellant(appeal.getAppellant());
+
+                        if (rpcEpimsId != null
+                            && courtVenue.getRegionId() != null) {
+                            callback.getCaseDetails().getCaseData()
+                                .setCaseManagementLocation(CaseManagementLocation.builder()
+                                    .baseLocation(rpcEpimsId)
+                                    .region(courtVenue.getRegionId())
+                                    .build());
+                        }
                     }
                 }
             }
```

Comment:
Nested if's like this makes it hard to understand.
Refactor to reduce the nested's to no more than 2 deep